### PR TITLE
Pandas examples

### PR DESCRIPTION
**Pull request by emdupre**
In response to #1140, updates examples to use pandas instead of numpy when handling csv files. Also patches `space_net.py` to check if input is classifier when converting to numeric, otherwise stimuli labels passed from pandas introduce errors.

### CODE REVIEW

File: examples/02_decoding/plot_haxby_anova_svm.py [hunk 1]
```diff
@@ -24,16 +24,18 @@
 # Load the behavioral data
 # -------------------------
 import numpy as np
```
Comment (GaelVaroquaux) [inline examples/02_decoding/plot_haxby_anova_svm.py]: Do we still need numpy in this example?

File: examples/02_decoding/plot_haxby_grid_search.py [hunk 2]
```diff
@@ -46,12 +47,12 @@
 print('Functional nifti image (4D) are located at: %s' % haxby_dataset.func[0])
 
 # Load the behavioral data
-labels = np.recfromcsv(haxby_dataset.session_target[0], delimiter=" ")
+labels = pd.read_csv(haxby_dataset.session_target[0], delimiter=" ")
```
Comment (GaelVaroquaux) [inline examples/02_decoding/plot_haxby_grid_search.py]: Should it be "sep" rather than "delimiter"?

File: examples/plot_decoding_tutorial.py [hunk 3]
```diff
@@ -184,9 +184,10 @@
 cv = KFold(n=len(fmri_masked), n_folds=5)
 
 for train, test in cv:
-    svc.fit(fmri_masked[train], conditions[train])
+    conditions_masked = conditions.values[train]
+    svc.fit(fmri_masked[train], conditions_masked)
     prediction = svc.predict(fmri_masked[test])
-    print((prediction == conditions[test]).sum() / float(len(conditions[test])))
+    print((prediction == conditions[test]).sum() / float(len(conditions.values[test])))
```
Comment (GaelVaroquaux) [inline examples/plot_decoding_tutorial.py]: Strange: you use ".values" in one of the indexing to "conditions" but not the other.

File: examples/04_manipulating_images/plot_roi_extraction.py [hunk 4]
```diff
@@ -307,7 +308,7 @@
 # Preparing for data extraction: setting number of conditions, size, etc from
 # haxby dataset
 condition_names = list(set(haxby_labels))
```
Comment (GaelVaroquaux) [inline examples/04_manipulating_images/plot_roi_extraction.py]: Can we replace the above by "haxby_labels.unique()"?

File: examples/04_manipulating_images/plot_roi_extraction.py [hunk 5]
```diff
@@ -307,7 +308,7 @@
...
-n_cond_img = fmri_data[..., haxby_labels == b'house'].shape[-1]
+n_cond_img = fmri_data[..., haxby_labels == 'house'].shape[-1]
```
Comment (GaelVaroquaux) [inline examples/04_manipulating_images/plot_roi_extraction.py]: Would "(haxby_labels == 'house').sum()" give the same result?

File: examples/02_decoding/plot_haxby_anova_svm.py (hunk 1 above)
Comment (emdupre) [reply examples/02_decoding/plot_haxby_anova_svm.py]: It's removed, and in `plot_haxby_space_net.py`!

File: examples/02_decoding/plot_haxby_grid_search.py (hunk 2 above)
Comment (emdupre) [reply examples/02_decoding/plot_haxby_grid_search.py]: Patched this for consistency, thanks!

File: examples/04_manipulating_images/plot_roi_extraction.py (hunk 5 above)
Comment (emdupre) [reply examples/04_manipulating_images/plot_roi_extraction.py]: This gives a different result, actually, so I'm keeping with the same code for now!

File: examples/05_advanced/plot_haxby_mass_univariate.py [hunk 6]
```diff
@@ -52,14 +52,15 @@
 ##############################################################################
 # Restrict to faces and houses
 import numpy as np
-labels = np.recfromcsv(haxby_dataset.session_target[0], delimiter=" ")
+import pandas as pd
+labels = pd.read_csv(haxby_dataset.session_target[0], delimiter=" ")
```
Comment (bthirion) [inline examples/05_advanced/plot_haxby_mass_univariate.py]: sep=" "

File: nilearn/decoding/space_net.py [hunk 7]
```diff
@@ -720,7 +720,7 @@ def fit(self, X, y):
         X = self.masker_.fit_transform(X)
 
         X, y = check_X_y(X, y, ['csr', 'csc', 'coo'], dtype=np.float,
-                         multi_output=True, y_numeric=True)
+                         multi_output=True, y_numeric=not self.is_classif)
```
Comment (bthirion) [inline nilearn/decoding/space_net.py]: Seems unrelated, but correct.

File: examples/05_advanced/plot_haxby_mass_univariate.py (hunk 6 above)
Comment (emdupre) [reply examples/05_advanced/plot_haxby_mass_univariate.py]: Thanks for catching that!

**Comment by emdupre** [comment]
I hadn't seen that @GaelVaroquaux opened a new PR on this, but I made commits addressing the doc build failure and moving numpy imports to relevant cells. I can close this, though, if you'd rather move ahead with #1562!

**Comment by GaelVaroquaux** [comment]
I think that it's better if you just integrate the relevant changes from my PR and address the comments that have been made on it

⁣Sent from TypeApp ​

**Comment by emdupre** [comment]
Pushed the bug fix you implemented, and 4552ba7 should address the review comments on the other PR

File: nilearn/decoding/space_net.py (hunk 7 above)
Comment (GaelVaroquaux) [reply nilearn/decoding/space_net.py]: It's actually not unrelated: the changes that we did implied that we were passing in non numerical values for y, and hence the code started failing where it shoudn't :).

**Comment by GaelVaroquaux** [comment]
OK, this looks great. I am merging this and adding an entry to doc/whats_new.rst
